pkg/agent/plugin/workloadattestor/windows/windows_windows_test.go

```diff
@@ -0,0 +1,276 @@
+		},
+		{
+			name: "OpenProcessToken error",
+			pq: &fakeProcessQuery{
+				openProcessTokenErr: errors.New("open process token error"),
+				handle:              windows.InvalidHandle,
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to open the access token associated with the process: open process token error",
+		},
+		{
+			name: "GetTokenUser error",
+			pq: &fakeProcessQuery{
+				getTokenUserErr: errors.New("get token user error"),
+				handle:          windows.InvalidHandle,
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to retrieve user account information from access token: get token user error",
+		},
+		{
+			name: "GetTokenGroups error",
+			pq: &fakeProcessQuery{
+				getTokenGroupsErr: errors.New("get token groups error"),
+				handle:            windows.InvalidHandle,
+				tokenUser:         &windows.Tokenuser{User: windows.SIDAndAttributes{Sid: sidUser}},
+			},
+			expectCode: codes.Internal,
+			expectMsg:  "workloadattestor(windows): failed to get process information: failed to retrieve group accounts information from access token: get token groups error",
+		},
+		{
+			name: "LookupAccount failure",
+			pq: &fakeProcessQuery{
+				lookupAccountErr: errors.New("lookup error"),
+				handle:           windows.InvalidHandle,
+				tokenUser:        &windows.Tokenuser{User: windows.SIDAndAttributes{Sid: sidUser}},
+				tokenGroups:      &windows.Tokengroups{Groups: [1]windows.SIDAndAttributes{{Sid: sidGroup1}}},
+				sidAndAttributes: []windows.SIDAndAttributes{{Sid: sidGroup1, Attributes: windows.SE_GROUP_ENABLED}},
+			},
+			selectorValues: []string{"user_sid:" + sidUser.String(), "enabled_group_sid:" + sidGroup1.String()},
+			expectCode:     codes.OK,
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.WarnLevel,
+					Message: "failed to lookup account from user SID",
+					Data: logrus.Fields{
+						"sid":           sidUser.String(),
+						logrus.ErrorKey: "lookup error",
+					},
+				},
+				{
+					Level:   logrus.WarnLevel,
+					Message: "failed to lookup account from group SID",
+					Data: logrus.Fields{
+						"sid":           sidGroup1.String(),
+						logrus.ErrorKey: "lookup error",
+					},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			defer test.logHook.Reset()
+
+			p := test.loadPlugin(t, testCase.pq)
+
+			selectors, err := p.Attest(ctx, 123)
+			spiretest.RequireGRPCStatus(t, err, testCase.expectCode, testCase.expectMsg)
+			if testCase.expectCode != codes.OK {
+				require.Nil(t, selectors)
+				return
+			}
+
+			require.NoError(t, err)
+			require.NotNil(t, selectors)
+			var selectorValues []string
+			for _, selector := range selectors {
```

Comment:
nit: what do you think about concat type+value and create a []string and compare it, 
In case we have a selector that is not "windows" it will found what selector is failing for example:
```
var selectorsValues []string
for _, selector := range selectors {
    selectorValues = append(selectorValues, selector.Type+selector.Value)
}

require.Equal(t, testCase.selectorValues, selectorValues)
```